### ISSUE TITLE
PredictiveTextInput accessibility

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#818]
**Component**
`PredictiveTextInput`

**Describe the feature request**
- Align the PredictiveTextInput with the accessibility behaviour defined in the Figma specs

**Acceptance criteria**
- Roles should be correctly read out by the AT
- Label should be correctly associated with the input field
- Close button should be read out by the AT
- Instructions for keyboard navigation should be read out by the AT
- Disabled, readonly state should be correctly read out by the AT
- Keyboard support is included

**Screenshots or Documentation**
[Figma](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=12-703&p=f&m=dev)
[PredictiveTextInput Storybook](https://designsystem.life.gov.sg/react/index.html?path=/docs/form-predictivetextinput--docs)
